storage/innobase/srv/srv0srv.cc

```diff
@@ -3390,3 +3392,48 @@ srv_is_undo_tablespace(
 	       && space_id < (srv_undo_space_id_start
 			      + srv_undo_tablespaces_open));
 }
+
+/** Gets page fragmentation statistics
+@param[out]	stats	a reference to fragmentation statistics to fill */
+void
+srv_get_fragmentation_stats(fragmentation_stats_t& stats)
+{
+	THD*	thd = current_thd;
+	if (UNIV_LIKELY(thd != NULL)) {
+		ulonglong	local_scan_pages_contiguous = 0;
+		ulonglong	local_scan_pages_disjointed = 0;
+		ulonglong	local_scan_pages_total_seek_distance = 0;
+		ulonglong	local_scan_data_in_pages = 0;
+		ulonglong	local_scan_garbage_in_pages = 0;
+		thd_get_fragmentation_stats(thd,
```

Comment:
This usage with sql/sql_thd_internal_api.cc change makes me think that the struct is better moved to server layer?

Reply:
Reworked